Postgres/Syntax/QueryDSL.lean

```diff
@@ -0,0 +1,105 @@
+/-
+  Copyright (c) 2022 Arthur Paulino. All rights reserved.
+  Released under Apache 2.0 license as described in the file LICENSE.
+  Authors: Arthur Paulino, Florian Würmseer
+-/
+
+import Postgres.Syntax.DataEntries
+
+inductive SQLSelectField
+  | col   : String → SQLSelectField
+  | alias : String → String         → SQLSelectField
+
+inductive SQLSelect
+  | list : Bool → List SQLSelectField → SQLSelect
+  | all  : Bool → SQLSelect
+
+inductive SQLProp
+  | tt : SQLProp
+  | ff : SQLProp
+  | eqC : String  → String    → SQLProp
+  | neC : String  → String    → SQLProp
+  | ltC : String  → String    → SQLProp
+  | leC : String  → String    → SQLProp
+  | gtC : String  → String    → SQLProp
+  | geC : String  → String    → SQLProp
+  | eqE : String  → DataEntry → SQLProp
+  | neE : String  → DataEntry → SQLProp
+  | ltE : String  → DataEntry → SQLProp
+  | leE : String  → DataEntry → SQLProp
+  | gtE : String  → DataEntry → SQLProp
+  | geE : String  → DataEntry → SQLProp
+  | and : SQLProp → SQLProp   → SQLProp
+  | or  : SQLProp → SQLProp   → SQLProp
+  | not : SQLProp → SQLProp
+
+inductive SQLJoin
+  | inner | left | right | outer
+
+inductive SQLFrom
+  | table        : String  → SQLFrom
+  | alias        : SQLFrom → String  → SQLFrom
+  | join         : SQLJoin → SQLFrom → SQLFrom → SQLProp → SQLFrom
+  | implicitJoin : SQLFrom → SQLFrom → SQLFrom
```

Comment:
I believe full SQL allows an arbitrary query here. I don't think you need to implement this now, but it will present interesting typechecking challenges later.